packages/zowe-explorer/src/trees/job/JobFSProvider.ts

```diff
@@ -214,18 +214,19 @@ export class JobFSProvider extends BaseProvider implements vscode.FileSystemProv
         const jesApi = ZoweExplorerApiRegister.getJesApi(spoolEntry.metadata.profile);
         await AuthHandler.waitForUnlock(spoolEntry.metadata.profile);
         try {
+            const spoolEncoding = spoolEntry.encoding?.kind === "other" ? spoolEntry.encoding.codepage : profileEncoding;
             if (jesApi.downloadSingleSpool) {
                 const spoolDownloadObject: IDownloadSpoolContentParms = {
                     jobFile: spoolEntry.spool,
                     stream: bufBuilder,
                     binary: spoolEntry.encoding?.kind === "binary",
-                    encoding: spoolEntry.encoding?.kind === "other" ? spoolEntry.encoding.codepage : profileEncoding,
+                    encoding: spoolEncoding,
```

Comment:
I know its unlikely, but if I'm pulling a spool file as binary and I have an encoding specified on my profile, won't it pass both `binary` and `encoding` options? I'm curious to know if this affects the behavior of the API, or if `binary` always takes precedence when specified on the API call

Reply:
Binary will take precedence over encoding in the z/OS Jobs SDK if both are specified:
```
        if (parms.binary) {
            parameters += "?mode=binary";
        } else if (parms.record) {
            parameters += "?mode=record";
        }

        if (!parms.binary && !parms.record && parms.encoding?.trim()) {
            parameters += "?fileEncoding=" + parms.encoding;
        }
```